bittensor/_dataset/dataset_impl.py

```diff
@@ -1,678 +1,772 @@
+
+        """ 
+        Async save of json for storing text hashes
         Args:
-            file_meta (dict of str: int):
-                Specify the details of the dataset in the format of {'Name': , 'Hash':}.
-            text (str): 
-                The string to save to the file.
-        
-        Returns:
-            text (str):
-                The text in the file.                
+            path (str, required):
+                Path of the loaded json
+            include_root (bool, optional):
+                Include self.data_dir as the prefix.
+                    - if True, ths meants shortens the batch and 
+                    specializes it to be with respect to the dataset's 
+                    root path which is in ./bittensor/dataset
+            default (dict, optional):
+                If there is a file not found, what is the default value.
+        Returns: 
+            obj (dict)
+                Object of the saved JSON as a dictionary.
         """
-        folder_path = os.path.expanduser(os.path.join(self.data_dir, file_meta['Folder']))
-        full_path = os.path.expanduser(os.path.join(self.data_dir, file_meta['Folder'], file_meta['Hash']))
-        if not os.path.exists(folder_path):
-            os.makedirs(folder_path)
-        try:
-            with open(full_path, mode = 'w+') as f:
-                f.write(text)
-                logger.success("Saved:".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
-            return True
         
-        except Exception as E:
-            logger.warning("Save failed:".ljust(20) + "<blue>{}</blue>".format(file_meta['Name']))
-            return False
+        # Include the root if true (self.data_dir).
+        if include_root:
+            if self.data_dir != path[:len(self.data_dir)]:
+                path = os.path.join(self.data_dir, path)
 
-    def get_text(self, file_meta):
-        r""" Either load a file from disk or download it from IPFS
-        Args:
-            file_meta (dict of str: int):
-                Specify the details of the file in the format of {'Name': , 'Hash':}.
+        # Ensure extension.
+        dir_path = os.path.dirname(path)
+        if os.path.splitext(path)[-1] != '.json':
+            path += '.json'
+
+        # Ensure dictionary.
+        if not os.path.isdir(dir_path):
+            os.makedirs(dir_path)
 
-        Return:
-            text (str):
-                The text that we get from the file (from disk or IPFS).     
+        # Load default if file does not exist.
+        try:
+            with open(path, 'r') as f:
+                obj = json.load(f)
+        except FileNotFoundError:
+            obj = default
+        except json.JSONDecodeError:
+            obj = default
+        if isinstance(obj, str):
+            obj = json.loads(obj)
+
+        return obj
+
+    load_json = sync_wrapper(async_load_json)
```

Comment:
These function extensions should include a comment. It is not obvious, at least to me, what you are accomplishing with this.